src/librustc_driver/lib.rs

```diff
@@ -111,34 +111,48 @@ const BUG_REPORT_URL: &'static str = "https://github.com/rust-lang/rust/blob/mas
                                       md#bug-reports";
 
 #[inline]
-fn abort_msg(err_count: usize) -> String {
+fn abort_msg(err_count: usize) -> Option<String> {
     match err_count {
-        0 => "aborting with no errors (maybe a bug?)".to_owned(),
-        1 => "aborting due to previous error".to_owned(),
-        e => format!("aborting due to {} previous errors", e),
+        0 => Some("aborting with no errors (maybe a bug?)".to_owned()),
+        1 => None,
+        e => Some(format!("aborting due to {} previous errors", e)),
```

Comment:
It seems a little odd that we would issue a message due to multiple errors but not one error. Not sure how important consistency is here. Personally, I don't think either message is particularly useful - I only care about whether there are errors or not.